src/services/portProvider.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as netstat from 'node-netstat';
+
+netstat.commands.darwin = {
+    cmd: 'lsof',
+    args: ['-Pn', '-i4', '-sTCP:LISTEN']
+};
+
+// NOTE: The TS definition incorrectly asserts that `darwin` is const.
+// TODO: Update the TS definitions.
+(netstat.parsers.darwin as unknown) = (line: string, callback: (item: netstat.ParsedItem) => void) => {
+    const parts = line.split(/\s/).filter(String);
+    if (!parts.length || (parts.length != 9 && parts.length != 10 )) {
```

Comment:
I'm confused about why parts of length 9 or 10 are an issue?
Maybe add a comment to elaborate?

Reply:
I was following the pattern of the library's existing parsers; we just expect either 9 or 10 columns of data and bail if not found.  I've added a comment.